mautic/mautic
# Manage Git-controlled Mautic update with shell scripts

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Yes
| Related user documentation PR URL | Coming
| Related developer documentation PR URL | Coming
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/2672
| BC breaks? | No
| Deprecations? | No

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Shell scripts to update your Git instance with less effort.

#### Steps to test this PR:
* Please run your standard backup process before following these steps in any environment! *
1. From an older version of mautic (currently non 2.2.0 release), run the git-stage-update.sh with `./scripts/git-stage-update.sh`
2. Then deploy your `staging` branch to your staging or production environment and run `./scripts/apply-update.sh`.
3. Confirm and test the update applied cleanly. If you had issues, you may need to rollback, in which case you would just checkout your `master` branch again and fix `staging` separately (or wait until the next release).
4. Once you have confirmed the updates are working, run `./scripts/git-finalize-update.sh`
